### ISSUE TITLE
react-a11y-image-button-has-alt fails if type is an expression

### ISSUE THREAD

**Issue by mfedderly** [Microsoft/tslint-microsoft-contrib#476]
react-a11y-image-button-has-alt rule fails on inputs like this:

```
const type = "text";
return <input type={type} />;
```

Stack trace:
```
TypeError: Cannot read property 'toLowerCase' of undefined
    at ReactA11yImageButtonHasAltWalker.validateOpeningElement (/code/node_modules/tslint-microsoft-contrib/reactA11yImageButtonHasAltRule.js:68:77)
    at ReactA11yImageButtonHasAltWalker.visitJsxSelfClosingElement (/code/node_modules/tslint-microsoft-contrib/reactA11yImageButtonHasAltRule.js:57:14)
    at ReactA11yImageButtonHasAltWalker.SyntaxWalker.visitNode (/code/node_modules/tslint/lib/language/walker/syntaxWalker.js:413:22)
    at /code/node_modules/tslint/lib/language/walker/syntaxWalker.js:535:63
    at visitNode (/code/node_modules/typescript/lib/typescript.js:14800:24)
    at Object.forEachChild (/code/node_modules/typescript/lib/typescript.js:15073:24)
    at ReactA11yImageButtonHasAltWalker.SyntaxWalker.walkChildren (/code/node_modules/tslint/lib/language/walker/syntaxWalker.js:535:12)
    at ReactA11yImageButtonHasAltWalker.SyntaxWalker.visitReturnStatement (/code/node_modules/tslint/lib/language/walker/syntaxWalker.js:222:14)
    at ReactA11yImageButtonHasAltWalker.SyntaxWalker.visitNode (/code/node_modules/tslint/lib/language/walker/syntaxWalker.js:476:22)
    at /code/node_modules/tslint/lib/language/walker/syntaxWalker.js:535:63
```

**Comment by JoshuaKGoldberg** [comment]
Glad this is fixed, thanks!

We should note that #461 would have caught this.
